src/FileNavigator.java

```diff
@@ -0,0 +1,61 @@
+import java.util.*;
+
+public class FileNavigator {
+    private Map<String, List<FileData>> files;
+
+    public FileNavigator() {
+        this.files = new HashMap<>();
+    }
+
+    public void add(FileData fileData) {
+        if (!fileData.getPath().equals(getPathKey(fileData.getPath()))) {
+            System.out.println("Error: Path-key and file path do not match");
+            return;
+        }
+
+        if (!files.containsKey(fileData.getPath())) {
+            files.put(fileData.getPath(), new ArrayList<>(Collections.singletonList(fileData)));
+        } else {
+            List<FileData> fileList = files.get(fileData.getPath());
+            fileList.add(fileData);
+            files.put(fileData.getPath(), fileList);
+        }
+    }
+
+    public List<FileData> find(String path) {
+        if (files.containsKey(path)) {
+            return files.get(path);
+        }
+        return null;
+    }
+
+    public List<FileData> filterBySize(int size) {
```

Comment:
What if `size` <= 0? What should your method return in this case?
Moreover, rename `size` to `maxFileSize` as the proper naming gives more context to what the method should do and what the parameter is.